session.go

```diff
@@ -699,9 +720,22 @@ func (s *session) handleUnpackedPacket(packet *unpackedPacket, rcvTime time.Time
 
 	r := bytes.NewReader(packet.data)
 	var isAckEliciting bool
+	var containsValidFrames bool
+	var containsIgnoredFrames bool
+	var ignorePacket bool
+
+parseLoop:
 	for {
 		frame, err := s.frameParser.ParseNext(r, packet.encryptionLevel)
 		if err != nil {
+			if s.config.AttackTimeout > 0 && packet.encryptionLevel != protocol.Encryption1RTT {
+				errStr := err.Error()
+				if strings.Contains(errStr, "FRAME_ENCODING_ERROR") {
```

Comment:
Conditioning on the error string is not ideal.  Your type assertion for `handleUnpackedPacket` is a better model, even if that needs changes to `FrameParser`.